### ISSUE TITLE
False positive in elision lint with `'static`

### ISSUE THREAD

**Issue by Manishearth** [Manishearth/rust-clippy#171]
```
    fn parse_error(a:&mut u8, msg: &'static str) {
    }
```

should not trigger the lint.

I guess we should be ignoring `'static`? I'm not sure what we should do here

cc @birkenfeld
